fedimint-server/src/consensus/server.rs

```diff
@@ -452,13 +452,26 @@ impl ConsensusServer {
         override_proposal: Option<ConsensusProposal>,
     ) -> ConsensusProposal {
         while let Some(Some(event)) = self.api_receiver.next().now_or_never() {
-            self.consensus.api_event_cache.insert(event);
+            match event {
+                ApiEvent::ForceProcessOutcome(outcome) => self.force_process_epoch(outcome).await,
+                event => {
+                    self.consensus.api_event_cache.insert(event);
+                }
+            };
         }
         let consensus_proposal = self.consensus.get_consensus_proposal().await;
         self.consensus.api_event_cache.clear();
         override_proposal.unwrap_or(consensus_proposal)
     }
 
+    async fn force_process_epoch(&mut self, outcome: EpochOutcome) {
+        let convert = ConsensusOutcomeConversion::from(outcome).0;
+        match self.process_outcome(convert).await {
+            Ok(_) => {}
+            Err(err) => warn!("Unable to force process epoch {:?}", err),
+        }
+    }
+
```

Comment:
Realized we already handle that here https://github.com/fedimint/fedimint/blob/master/fedimint-server/src/consensus/server.rs#L334-L335

If an epoch is in the past, it won't be processed, if it's in the future we'll download missing epochs.

Reply:
Added a test for it though.

Reply:
I don't think this fixes the problem, this comment wasn't about downloading old epochs.

Let's suppose there `f` servers are permanently lost and `t` (A, B, C) are trying to restart but one of them (A) is at a wrong epoch:

* A, B, C start HBBFT at their respective last epochs
* A, B, C send rejoin requests and generate HBBFT messages
  * A sends message for epoch `n`
  * B, C send messages for epoch `n+1`
  * No consensus is found on the epoch, A stays on epoch `n`
* No progress is made since only 2 of the required 3 are sending contributions for the correct epoch
* A forces the processing of epoch `n+1`, **but does not tell HBBFT about it, so no new messages are generated for epoch `n+1` and HBBFT remains stuck**

I think this is solved by restarting `fedimintd` afterwards, but ideally there'd be a more elegant solution.